neqo-transport/src/connection.rs

```diff
@@ -1915,10 +1989,18 @@ mod tests {
         let mut a = client;
         let mut b = server;
         let mut datagram = None;
-        let is_done = |c: &mut Connection| match c.state() {
-            // TODO(mt): Finish on Closed and not Closing.
-            State::Connected | State::Closing { .. } | State::Closed(..) => true,
-            _ => false,
+        let mut more_iters = 1;
+        let mut is_done = |c: &mut Connection| {
+            let done_state = match c.state() {
+                // TODO(mt): Finish on Closed and not Closing.
+                State::Connected | State::Closing { .. } | State::Closed(..) => true,
+                _ => false,
+            };
+            if done_state && more_iters != 0 {
+                more_iters -= 1;
+                return false;
+            }
+            done_state
```

Comment:
I take it that you are trying to ensure that you run one more iteration of the loop after both endpoints complete the handshake so that the ACK can get through.  Why not instead do:

```rust
let mut datagram = None;
while !is_done(a) {
  datagram = a.process(datagram, now()).dgram();
  mem::swap(&mut a, &mut b);
}
a.process(datagram, now());
```

Reply:
so much better.